File: 4256-construct-uniform-parity-array-i/construct-uniform-parity-array-i.py
class Solution(object):
    def uniformArray(self, nums1):
        oddcount=0
        evencount=0

        for num in nums1:
            if num%2==0:
                evencount+=1
            else:
                oddcount+=1

        if evencount==len(nums1):
            return(True)
        if oddcount>0:
            return(True)
        return False
        

File: 4256-construct-uniform-parity-array-i/construct-uniform-parity-array-i.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        
        int oddCount = 0;
        int evenCount = 0;

        for (int num : nums1) {
            if (num % 2 == 0) {
                evenCount++;
            } else {
                oddCount++;
            }
        }

        // If all numbers are even, keep them as they are.
        if (evenCount == nums1.length) {
            return true;
        }

        // If there is at least one odd number,
        // we can make all elements odd:
        // - Keep odd numbers unchanged.
        // - For every even number, subtract an odd number.
        //   even - odd = odd.
        if (oddCount > 0) {
            return true;
        }

        return false;
    }
}